File: analytics/src/test/java/io/appmetrica/analytics/impl/utils/validation/NonNullValidatorTest.java
package io.appmetrica.analytics.impl.utils.validation;

import io.appmetrica.analytics.testutils.CommonTest;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

public class NonNullValidatorTest extends CommonTest {

    private final Validator mValidator = new NonNullValidator("test non null");

    @Test
    public void testValid() {
        assertThat(mValidator.validate(mock(Object.class)).isValid()).isTrue();
    }

    @Test
    public void testNonValid() {
        assertThat(mValidator.validate(null).isValid()).isFalse();
    }

    @Test
    public void testNonValidDescription() {
        assertThat(mValidator.validate(null).getDescription()).startsWith("test non null");
    }

}

File: core-utils/src/test/java/io/appmetrica/analytics/coreutils/internal/validation/NonNullValidatorTest.kt
package io.appmetrica.analytics.coreutils.internal.validation

import io.appmetrica.gradle.testutils.CommonTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.mockito.kotlin.mock

class NonNullValidatorTest : CommonTest() {

    private val validator = NonNullValidator<Any>("test non null")

    @Test
    fun nonNullPasses() {
        assertThat(validator.validate(mock()).isValid).isTrue
    }

    @Test
    fun nullFails() {
        val result = validator.validate(null)
        assertThat(result.isValid).isFalse
        assertThat(result.description).startsWith("test non null")
    }
}
